sdk/monitor/Azure.Monitor.OpenTelemetry.LiveMetrics/src/Internals/Manager.Metrics.cs

```diff
@@ -149,5 +164,70 @@ public MonitoringDataPoint GetDataPoint()
 
             return s_isAzureWebApp;
         }
+
+        private void ResetCachedValues()
+        {
+            this.cachedCollectedTime = DateTimeOffset.MinValue;
+            this.cachedCollectedValue = 0;
+        }
+
+        /// <summary>
+        /// Calcualte the CPU usage as the diff between two ticks divided by the period of time, and then divided by the number of processors.
+        /// </summary>
+        private bool TryCalculateCPUCounter(Process process, out double normalizedValue)
```

Comment:
Why can't we use the same code as in process instrumentation - https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/de135415f5de21c0a565f3fb3b0570484e7a2bae/src/OpenTelemetry.Instrumentation.Process/ProcessMetrics.cs#L41?

Reply:
The Process Instrumentation is not appropriate for our purposes.

1. The instrumentation is intending to record a "monotonically increasing value", that is a value that only increases.
  https://opentelemetry.io/docs/specs/otel/metrics/api/#asynchronous-counter The OpenTelemetry doc actually specifies CPU Time as an example. It's recording the ticks that the process is utilizing the CPU. This value will only ever increase.

    I tested reporting values to LiveMetrics in this way and that results in this chart: 
    Note that % Total increases past 100 during the stress test, but then flatlines when the stress test completes.
    ![image](https://github.com/Azure/azure-sdk-for-net/assets/28785781/60c344c0-1f8c-40a4-a368-e995d5dfeee8)


2. LiveMetrics is intending to show CPU utilization %. This value is expected to increase when under load, but then decrease back to zero. The Process counters only provide a "monotonically increasing" CPU tick. So we need to calculate the change over time.

    `((change in ticks / period) / number of processes)`
    
    ![image](https://github.com/Azure/azure-sdk-for-net/assets/28785781/3bce58c1-4727-445a-884c-57985ec918b3)


3. The OpenTelemetry instrumentation is reporting two counters separately (`process.UserProcessorTime` and `process.PrivilegedProcessorTime`). LiveMetrics doesn't need these separate values so I'm using `process.TotalProcessorTime` which is defined as [the sum of the UserProcessorTime and the PrivilegedProcessorTime.](
https://learn.microsoft.com/dotnet/api/system.diagnostics.process.totalprocessortime)


4. The OpenTelemetry instrumentation is recording `TotalSeconds`. I'm following AI SDK and using `Ticks`. I want to have the greatest possible precision in our calculations.